{
  SignalThrowingEventListener listener=null;
  try {
    listener=new SignalThrowingEventListener();
    listener.setSignalName("Signal");
    listener.setProcessInstanceScope(true);
    processEngineConfiguration.getEventDispatcher().addEventListener(listener,ActivitiEventType.JOB_EXECUTION_FAILURE);
    ProcessInstance processInstance=runtimeService.startProcessInstanceByKey("testSignal");
    assertNotNull(processInstance);
    Job signalJob=managementService.createJobQuery().processInstanceId(processInstance.getId()).singleResult();
    try {
      managementService.executeJob(signalJob.getId());
      fail("Exception expected");
    }
 catch (    ActivitiException ae) {
    }
    Job failedJob=managementService.createJobQuery().withException().processInstanceId(processInstance.getId()).singleResult();
    assertNotNull(failedJob);
    assertEquals(2,failedJob.getRetries());
    assertEquals(0,taskService.createTaskQuery().processInstanceId(processInstance.getId()).count());
    try {
      managementService.executeJob(failedJob.getId());
      fail("Exception expected");
    }
 catch (    ActivitiException ae) {
      assertEquals(0,taskService.createTaskQuery().processInstanceId(processInstance.getId()).count());
    }
  }
  finally {
    processEngineConfiguration.getEventDispatcher().removeEventListener(listener);
  }
}
